src/VirtoCommerce.EventBusModule.Data/Services/DefaultEventBusSubscriptionsManager.cs

```diff
@@ -77,9 +77,13 @@ protected virtual async Task HandleEvent(DomainEvent domainEvent, CancellationTo
 
             if (searchResult.TotalCount > 0)
             {
-                var events = domainEvent.GetEntityWithInterface<IEntity>()
+                var entities = domainEvent.GetEntityWithInterface<IEntity>()
                                              .Select(x => new EventData { ObjectId = x.Id, ObjectType = x.GetType().FullName, EventId = eventId })
                                              .ToArray();
+                var valueObjects = domainEvent.GetEntityWithInterface<IValueObject>()
```

Comment:
maybe use <ValueObject> instead of <IValueObject>?  since we explicitly cast to ValueObject below

Reply:
then need to change the extension
https://github.com/VirtoCommerce/vc-module-event-bus/blob/dev/src/VirtoCommerce.EventBusModule.Core/Extensions/DomainEventExtensions.cs#L22
perhaps will have issues how to compare correct classes
the code just a faster fix